src/script/cpp_api/s_base.cpp

```diff
@@ -121,6 +121,14 @@ ScriptApiBase::ScriptApiBase(ScriptingType type):
 	lua_newtable(m_luastack);
 	lua_setglobal(m_luastack, "core");
 
+	// vector.metatable is stored in the registry for quick access from C++.
+	lua_newtable(m_luastack);
+	lua_rawseti(m_luastack, LUA_REGISTRYINDEX, CUSTOM_RIDX_VECTOR_METATABLE);
+	lua_newtable(m_luastack);
+	lua_rawgeti(m_luastack, LUA_REGISTRYINDEX, CUSTOM_RIDX_VECTOR_METATABLE);
+	lua_setfield(m_luastack, -2, "metatable");
+	lua_setglobal(m_luastack, "vector");
```

Comment:
The part I don't like about this is that `vector` stays in the insecure env.
(see also https://github.com/minetest/minetest/commit/9a12e4499ecf5c1a3467af9c831d0d350a21923d#diff-f65d96a13c649554a5df1643c51d66874c21ff698642949cf3b618b4ba3c68f5R252)